retail/interactive-tutorial/RetailEvents.Samples/ImportUserEventsGcsSample.cs

```diff
@@ -0,0 +1,124 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_import_user_events_from_gcs]
+// Import user events into a catalog from GCS using Retail API
+
+using Google.Cloud.Retail.V2;
+using Google.LongRunning;
+using System;
+
+/// <summary>
+/// The import user events GCS sample class.
+/// </summary>
+public class ImportUserEventsGcsSample
+{
+    /// <summary>
+    /// Get import user events from GCS request.
+    /// </summary>
+    /// <param name="gcsObjectName">The name of the GCS object that contains the list of user events in JSON format.</param>
+    /// <param name="projectId">The current project id.</param>
+    /// <returns>The import user events request.</returns>
+    private static ImportUserEventsRequest GetImportUserEventsGcsRequest(string gcsObjectName, string projectId, string eventsBucketName)
+    {
+        string gcsBucket = $"gs://{eventsBucketName}";
+        string gcsErrorsPrefix = $"{gcsBucket}/error";
+
+        string locationId = "global";
+        string catalogId = "default_catalog";
+        // To check error handling paste the invalid catalog name here:
+        // catalogId = "invalid_catalog_name";
+        CatalogName defaultCatalog = new CatalogName(projectId, locationId, catalogId);
+
+        GcsSource gcsSource = new GcsSource();
+        gcsSource.InputUris.Add($"{gcsBucket}/{gcsObjectName}");
+
+        Console.WriteLine("GCS source:");
+        Console.WriteLine(gcsSource.InputUris);
+        Console.WriteLine();
+
+        ImportUserEventsRequest importRequest = new ImportUserEventsRequest
+        {
+            ParentAsCatalogName = defaultCatalog,
+            InputConfig = new UserEventInputConfig
+            {
+                GcsSource = gcsSource
+            },
+            ErrorsConfig = new ImportErrorsConfig
+            {
+                GcsPrefix = gcsErrorsPrefix
+            }
+        };
+
+        Console.WriteLine("Import user events from google cloud source request:");
+        Console.WriteLine(importRequest);
+        Console.WriteLine();
+
+        return importRequest;
+    }
+
+    /// <summary>
+    /// Call the Retail API to import user events.
+    /// </summary>
+    /// <param name="projectId">The current project id.</param>
+    public static Operation<ImportUserEventsResponse, ImportMetadata> ImportUserEventsFromGcs(string projectId, string eventsBucketName)
+    {
+        string gcsEventsObject = "user_events.json";
+        // To check error handling use the json with invalid user event:
+        // string gcsEventsObject = "user_events_some_invalid.json";
```

Comment:
So that they don't get a build error if they uncomment the entire line.
```suggestion
        // gcsEventsObject = "user_events_some_invalid.json";
```

Reply:
I should create another PR to solve this issue.

Reply:
#1855